src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -50,49 +50,14 @@ internal class RazorLanguageServerClient : ILanguageClient, ILanguageClientCusto
 
         [ImportingConstructor]
         public RazorLanguageServerClient(
-            RazorLanguageServerCustomMessageTarget customTarget,
-            RazorLanguageClientMiddleLayer middleLayer,
-            LSPRequestInvoker requestInvoker,
-            ProjectConfigurationFilePathStore projectConfigurationFilePathStore,
-            RazorLanguageServerLogHubLoggerProviderFactory logHubLoggerProviderFactory,
-            VSLanguageServerFeatureOptions vsLanguageServerFeatureOptions,
+            RazorLanguageServerCustomMessageTarget customTarget!!,
+            RazorLanguageClientMiddleLayer middleLayer!!,
+            LSPRequestInvoker requestInvoker!!,
+            ProjectConfigurationFilePathStore projectConfigurationFilePathStore!!,
+            RazorLanguageServerLogHubLoggerProviderFactory logHubLoggerProviderFactory!!,
+            VSLanguageServerFeatureOptions vsLanguageServerFeatureOptions!!,
             [Import(AllowDefault = true)] VisualStudioHostServicesProvider? vsHostWorkspaceServicesProvider)
         {
-            if (customTarget is null)
-            {
-                throw new ArgumentNullException(nameof(customTarget));
-            }
-
-            if (middleLayer is null)
-            {
-                throw new ArgumentNullException(nameof(middleLayer));
-            }
-
-            if (requestInvoker is null)
-            {
-                throw new ArgumentNullException(nameof(requestInvoker));
-            }
-
-            if (projectConfigurationFilePathStore is null)
-            {
-                throw new ArgumentNullException(nameof(projectConfigurationFilePathStore));
-            }
-
-            if (logHubLoggerProviderFactory is null)
-            {
-                throw new ArgumentNullException(nameof(logHubLoggerProviderFactory));
-            }
-
-            if (vsLanguageServerFeatureOptions is null)
-            {
-                throw new ArgumentNullException(nameof(vsLanguageServerFeatureOptions));
-            }
-
-            if (vsHostWorkspaceServicesProvider is null)
-            {
-                throw new ArgumentNullException(nameof(vsHostWorkspaceServicesProvider));
-            }
```

Comment:
😕 Why are we removing this line instead of making the argument required?

Reply:
📝 Note that null checking an optional MEF import is intentional in some cases:
https://github.com/dotnet/roslyn/pull/59047

Reply:
I'm confused.. when you say "make the argument required" do you mean remove the `AllowDefault`? If so, well because I assume the MEF import sometime isn't satisfied, and if not, well that just seems counter intuitive.

Reply:
Gonna merge this, as with so many changes its a merge conflict waiting to happen. Will follow up if there is any action here.

Reply:
The desired outcome here is to continue to have `[Import(AllowDefault = true)]`, but also have `!!` on the parameter. This results in a MEF 2 composition which will not fail to compose for lack of this part, but will throw an exception if anything tries to obtain this part. For example, if the catalog doesn't provide `vsHostWorkspaceServicesProvider`, it would be perfectly acceptable to obtain `Lazy<RazorLanguageServerClient>` from the catalog, but you could not access `Lazy<RazorLanguageServerClient>.Value` without an exception.

Reply:
➡️ Filed follow-up issue https://github.com/dotnet/razor-tooling/issues/6165

Reply:
Interesting.. and weird :P

Does it change your thinking at all knowing that the code that uses the service later was already defensive against nulls?